db/index.js

```diff
@@ -30,7 +30,9 @@ let initOptions = {
 	// API: http://vitaly-t.github.io/pg-promise/global.html#event:extend
 	extend: object => {
 		Object.keys(repos).forEach(repoName => {
-			object[repoName] = new repos[repoName](object, pgp);
+			if (!!repos[repoName].prototype && !!repos[repoName].prototype.constructor.name) {
+				object[repoName] = new repos[repoName](object, pgp);
+			}
```

Comment:
@karmacoma As per my knowledge, when ever a transaction/task context is initiated, that extend method is called over with that `object` it to extend it with repositories. So we should avoid additional conditions inside this method and instead we can update line#20 `require('require-all')(`${__dirname}/repos`);` to explicits required all database repositories.

Reply:
Or more better approach can be to create a `index.js` file which will require all files and export. 

```
modules.export = {
    accounts: require('./accounts.js'),
    delegates: require('./delegates.js'),
    .....
}
```

Reply:
@nazarhussain I think the filtering of require-all can be done, but is not sufficient. Hence why I am performing a basic check on the module interface before instantiating anything. If the modules shared a common property/function on the prototype, I would also check for that.

Yes static `index.js` file would work also, but I guess I was conforming to the existing require all approach in order to reduce maintenance.

Reply:
I don't see continuous update to such `index.js` file because its least likely that we will grow our database repositories so often. So creating that file is just a one time effort and will avoid any conditioning and filtering in extending transaction/task object.

Reply:
Ok, you've convinced me. We will go with the static route and remove this `require-all` dependency also while we are at it.

Reply:
Good move, guys. This is exactly what I did in the demo, by the way :wink: https://github.com/vitaly-t/pg-promise-demo/blob/master/JavaScript/db/repos/index.js